File: iOS/BalloonChatView/BalloonChatView/Model/ChatMessage.swift
//
//  ChatMessage.swift
//  BalloonChatView
//
//  Created by yuppe on 2023/01/11.
//

import Foundation

struct ChatMessageItem: Identifiable {
    var id = UUID()
    var date = Date()
    var from: User
    var to: User
    var text: String
}

class ChatMessage: ObservableObject {
    @Published var items: [ChatMessageItem] = []
    
    func append(_ item: ChatMessageItem) {
        self.items.append(item)
    }
}

File: Android/BalloonChatView/app/src/main/java/com/example/balloonchatview/view/model/ChatMessage.kt
package com.example.balloonchatview.view.model

import java.util.*

class ChatMessage(from: User, to: User, text: String) {
    var id = UUID.randomUUID()
    var date = Date()
    var from: User
    var to: User
    var text: String
    var readCount: Int = 1

    init {
        this.from = from
        this.to = to
        this.text = text
    }
}
